client/app/dbaas/dbaas-metrics/token/metrics-token.html

```diff
@@ -5,45 +5,48 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
-                data-ng-model-options="{ debounce: 400 }"
-                data-translate-attr="{ placeholder: 'metrics_token_search' }"/>
-        </div>
-    </div>
-    <div class="metrics__row">
-        <oui-table id="token-table"
-                   rows="MetricsTokenCtrl.tokens"
-                   rows-loader="MetricsTokenCtrl.loading">
-            <column title="'metrics_table_name' | translate"
-                    property="description"
-                    sortable>
-                <span data-ng-bind="$row.description || '-'"></span>
-            </column>
-            <column title="'metrics_token_labels' | translate"
-                    property="labels"
-                    template="MetricsTokenCtrl.labelsTemplate()">
-            </column>
-            <column title="'metrics_table_token' | translate"
-                    property="access">
-                <oui-clipboard data-model="$row.access"></oui-clipboard>
-            </column>
-            <column title="'metrics_token_permission' | translate"
-                    property="permission">
+        <oui-datagrid rows="MetricsTokenCtrl.tokens">
+            <extra-top>
+                <button data-ng-if="!MetricsTokenCtrl.loading"
+                    class="oui-button oui-button_secondary" type="button"
+                    data-ng-disabled="MetricsTokenCtrl.actions.loading"
+                    data-translate="metrics_token_add_token"
+                    data-ui-sref="dbaas.metrics.detail.token.add({ serviceName: MetricsTokenCtrl.serviceName })"></button>
+                <oui-spinner data-ng-if="MetricsTokenCtrl.loading"></oui-spinner>                
+            </extra-top>
+            <oui-column data-title=":: 'metrics_table_name' | translate" data-sortable="asc" data-searchable data-property="description" data-type="string">
+                    <span data-ng-bind="$row.description || '-'"></span>
+            </oui-column>
+            <oui-column data-title=":: 'metrics_token_labels' | translate" data-type="string" data-property="labels">
+                <span class="token-labels" data-ng-repeat="label in $row.labels | orderBy : 'key' track by $index" data-ng-if="$index < 2">
+                    <span data-ng-bind="label.key"></span>:<span data-ng-bind="label.value"></span>
+                </span>
+                <oui-dropdown data-ng-if="$row.labels.length > 2">
+                    <button type="button"
+                        class="token-labels token-labels__button"
+                        data-ng-bind="MetricsTokenCtrl.displayRemainingLabels($row.labels.length - 2)"
+                        oui-dropdown-trigger>
+                    </button>
+                    <oui-dropdown-content>
+                        <div class="token-labels-container">
+                            <span class="token-labels" data-ng-repeat="label in $row.labels | orderBy : 'key' track by $index" data-ng-if="$index >= 2">
```

Comment:
I'm not sure about what a `token-labels` looks like but why you didn't use `chips` components? (ref: http://master.ui-kit.ovh/#!/oui-angular/chips)

Reply:
👍

Reply:
`token-labels` was a local class. Have removed it and added `chips` component with some customisation. Please have a look